### ISSUE TITLE
Rescue is unintuitive in egg hunts

### ISSUE THREAD

**Issue by Alayan-stk-2** [supertuxkart/stk-code#3390]
In egg hunts, there is no lap counting, no good or wrong direction.

However, the evaluation of the correct position for rescuing still work as in normal races with lap.

For example, if you go in the reverse direction because an egg is closer that way, pressing rescue at some point will send you back to the start line.

**Comment by CodingJellyfish** [comment]
Is this duplicated with #605 ?

**Comment by Alayan-stk-2** [comment]
No, the other issue is about the conditions under which auto-rescue triggers. This issue is about where the kart is placed back on track after a rescue (be it triggered automatically or manually).
